### PR TITLE
[DNS] Adapter API changes and handlers for deep noise suppression

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Adapter API changes and handlers for deep noise suppression

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3811789
https://skype.visualstudio.com/SPOOL/_workitems/edit/3811792

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 1]
```diff
@@ -678,6 +678,20 @@ export interface CallAdapterCallOperations {
...
+  /* @conditional-compile-remove(DNS) */
+  /**
+   * Start the noise suppression effect.
+   *
+   * @beta
+   */
+  startNoiseSuppressionEffect(): Promise<void>;
+  /* @conditional-compile-remove(DNS) */
+  /**
+   * Start the noise suppression effect.
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:690]:
```suggestion
   * Stop the noise suppression effect.
```

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 2]
```diff
@@ -120,6 +122,8 @@ import { VideoBackgroundEffectsDependency } from '@internal/calling-component-bi
 import { CallSurvey, CallSurveyResponse } from '@azure/communication-calling';
 import { CallingSoundSubscriber } from './CallingSoundSubscriber';
 import { CallingSounds } from './CallAdapter';
+/* @conditional-compile-remove(DNS) */
+import { DeepNoiseSuppressionEffect } from '@azure/communication-calling-effects';
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:126]: How big is the file size for this? Wondering if this needs lazy loaded like the video background effects

**Review by mgamis-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts (hunk 2 above)
Comment (dmceachernmsft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:126]: i wonder if we should just do that so we have better decoupling from the calling effects package?

Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:128]: It is adding ~4MB to our bundle size, it is still under discussion if we want to enforce the dependency resolution or handle it internally. There is a task created separately to track it https://skype.visualstudio.com/SPOOL/_workitems/edit/3811815

File: packages/calling-component-bindings/src/handlers/createCommonHandlers.ts [hunk 3]
```diff
@@ -578,6 +584,30 @@ export const createDefaultCommonCallingHandlers = memoizeOne(
...
+    const onStartNoiseSuppressionEffect = async (audioEffects: AudioEffectsStartConfig): Promise<void> => {
+      const stream = call?.localAudioStreams.find((stream) => stream.mediaStreamType === 'Audio');
+      if (stream && audioEffects && audioEffects.noiseSuppression) {
+        console.log('start noise suppression- active effects', stream.feature(Features.AudioEffects).activeEffects);
+        const audioEffectsFeature = stream.feature(Features.AudioEffects);
+        const isNoiseSuppressionSupported = await audioEffectsFeature.isSupported(audioEffects.noiseSuppression);
+        if (isNoiseSuppressionSupported) {
+          return await audioEffectsFeature.startEffects(audioEffects);
+        } else {
+          throw new Error('Deep Noise Suppression is not supported on this platform.');
```
[the remark this answers is not in the document]
Comment (prabhjot-msft) [reply packages/calling-component-bindings/src/handlers/createCommonHandlers.ts:624]: I am following up on this with calling team, will update the error message accordingly.

[the remark this answers is not in the document]
Comment (prabhjot-msft) [reply packages/calling-component-bindings/src/handlers/createCommonHandlers.ts:624]: Checked with the calling team, we don't have any required hardware checks but it is only supported on chrome and edge, Chromium 126 and up
image

**Pull request closed without merging**